packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -254,17 +254,26 @@ export function usePasteHandler( props ) {
 }
 
 /**
- * Normalizes a given string of HTML to remove the Windows specific "Fragment" comments
- * and any preceeding and trailing whitespace.
+ * Normalizes a given string of HTML to remove the Windows-specific "Fragment"
+ * comments and any preceeding and trailing content.
  *
  * @param {string} html the html to be normalized
  * @return {string} the normalized html
  */
 function removeWindowsFragments( html ) {
-	const startReg = /.*<!--StartFragment-->/s;
-	const endReg = /<!--EndFragment-->.*/s;
+	const startStr = '<!--StartFragment-->';
+	const startIdx = html.indexOf( startStr );
+	if ( startIdx > -1 ) {
```

Comment:
I think we could return early here if no match found. Also I haven't tested in windows but this still keeps the white space, which was the original intend for this function. I think we could use a check for the next/previous char if is white space char and update the index we use to slice..